File: Facade/IContactService.vb
Imports BO
Public Interface IContactService
    Function GetContacts() As GetResponse(Of ContactBO)
    Function GetContactById(id As Integer) As GetResponse(Of ContactBO)
    Function GetContactsByIds(IdList As List(Of Integer)) As GetResponse(Of ContactBO)
    Function GetContactsForSearchList(searchterm As String) As GetResponse(Of SelectBO)
    Function InsertUpdate(bo As ContactBO) As Response
    Function Delete(ids As List(Of Integer)) As Response
    Function Delete(bos As List(Of ContactBO)) As Response

End Interface

File: FacadeCore/IContactService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;

namespace FacadeCore
{
    public interface IContactService
    {
        GetResponse<ContactBO> GetContacts();
        GetResponse<ContactBO> GetContactById(int id);
        GetResponse<ContactBO> GetContactsByIds(List<int> IdList);
        GetResponse<SelectBO> GetContactsForSearchList(string searchterm);
        Response InsertUpdate(ContactBO bo);
        Response Delete(List<int> ids);
        Response Delete(List<ContactBO> bos);
    }
}
